bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,254 @@
+
+        synchronized (petcareAPI) {
+            boolean requiresUpdate = false;
+            SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+            if (device != null) {
+                SurePetcareDeviceControl existingControl = device.control;
+                logger.debug("Old Curfew Control: {}", existingControl);
+
+                SurePetcareDeviceCurfew curfew = existingControl.curfewList.get(slot - 1);
+                logger.debug("channelUIDBase: {}", channelUIDBase);
+
+                switch (channelUIDBase) {
+                    case DEVICE_CHANNEL_CURFEW_ENABLED:
+                        if (command instanceof OnOffType) {
+                            logger.debug("old enables: {}, command: {}", curfew.enabled, command.equals(OnOffType.ON));
+                            if (curfew.enabled != command.equals(OnOffType.ON)) {
+                                requiresUpdate = true;
+                            }
+                            curfew.enabled = (command.equals(OnOffType.ON));
+                        }
+                        break;
+                    case DEVICE_CHANNEL_CURFEW_LOCK_TIME:
+                        if ((command instanceof StringType) && (command.toString().matches(CURFEW_TIME_REGEXP))) {
+                            logger.debug("old locktime: {}, new locktime: {}", curfew.lockTime, command);
+
+                            if (!(curfew.lockTime.equals(command.toString())) && curfew.enabled) {
+                                requiresUpdate = true;
+                            }
+                            curfew.lockTime = command.toString();
+                        } else {
+                            logger.warn("Incorrect curfew time format HH:mm: {}", command);
+                        }
+                        break;
+                    case DEVICE_CHANNEL_CURFEW_UNLOCK_TIME:
+                        if ((command instanceof StringType) && (command.toString().matches(CURFEW_TIME_REGEXP))) {
+                            if (!(curfew.unlockTime.equals(command.toString())) && curfew.enabled) {
+                                requiresUpdate = true;
+                            }
+                            curfew.unlockTime = command.toString();
+                        } else {
+                            logger.warn("Incorrect curfew time format HH:mm: {}", command);
+                        }
+                        break;
+                    default:
+                        break;
+                }
+                logger.debug("requiresUpdate: {}", requiresUpdate);
+
+                if (requiresUpdate) {
+                    try {
+                        petcareAPI.setCurfews(device, existingControl.curfewList);
+                        updateThingCurfews(device);
+                    } catch (SurePetcareApiException e) {
+                        logger.warn("Error from SurePetcare API. Can't update curfews for device {}", device);
```

Comment:
Can you validate whether you correctly update the thing status were needed and decreased verbosity of logging in line with https://www.openhab.org/docs/developer/guidelines.html#f-logging